app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -285,6 +285,21 @@ function uploadTypeFromFileName(fileName) {
   return isAnImage(fileName) ? 'image' : 'attachment';
 }
 
+function isGUID(value) {
+  return /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i.test(value);
+}
+
+function imageNameFromFileName(fileName) {
+  const split = fileName.split('.');
+  const name = split[split.length-2];
+
+  if (exports.isAppleDevice() && isGUID(name)) {
```

Comment:
Using `exports.isAppleDevice()` feels like a hack to me. Is there a better way to make this testable? It looks like I can't override `navigator.userAgent`